.github/workflows/ci.yaml

```diff
@@ -0,0 +1,34 @@
+# Copyright 2025 NVIDIA CORPORATION
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+name: CI Pipeline
+
+on:
+  push:
+    branches:
+      - "pull-request/[0-9]+"
+      - main
+      - release-*
+  
+jobs:
+  basic:
+    uses: ./.github/workflows/basic-checks.yaml
+
+  image:
+    uses: ./.github/workflows/image.yaml
+    needs:
+    - basic
+    secrets: inherit
+    with:
+      version: ${{ needs.basic.outputs.version }}
+      build_multi_arch_images: ${{ github.ref_name == 'main' || startsWith(github.ref_name, 'release-') }}
```

Comment:
Question (not a blocker): given that we've optimized the multi-arch builds in this repo to use `GOARCH`, should we not just drop this or set it to `true` all the time?